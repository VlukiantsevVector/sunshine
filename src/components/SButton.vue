<template>
  <button
    :type="type"
    :class="buttonClasses"
    :disabled="inactive"
    v-on="$listeners"
  >
    <span :class="contentClasses" tabindex="-1">
      <slot name="left" />
      <span class="flex-grow">
        <slot />
      </span>
      <slot name="right" />
    </span>
  </button>
</template>

<script lang="ts">
import Vue from 'vue';
import mixins from 'vue-typed-mixins';
import { ClassesMixin } from '../lib/utils';

export default mixins(ClassesMixin).extend({
  name: 's-button',
  inheritAttrs: true,

  props: {
    small: {
      type: Boolean,
      default: false,
    },

    primary: {
      type: Boolean,
      default: false,
    },

    danger: {
      type: Boolean,
      default: false,
    },

    inactive: {
      type: Boolean,
      default: false,
    },

    submit: {
      type: Boolean,
      default: false,
    },
  },

  computed: {
    type(): string {
      return this.submit ? 'submit' : 'button';
    },

    buttonClasses(): object {
      return this.classes({
        small: this.small,
        normal: !this.primary && !this.danger,
        primary: this.primary,
        danger: this.danger,
        inactive: this.inactive,
      });
    },

    contentClasses(): object {
      return this.classes('content', {
      });
    },
  },
});
</script>
