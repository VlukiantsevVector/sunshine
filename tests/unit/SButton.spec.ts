import { mount } from '@vue/test-utils';
import SButton from '@/components/SButton.vue';

describe('SButton', () => {
  it('renders normal button', () => {
    const wrapper = mount(SButton, {
      slots: {
        default: 'Normal button',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders primary button', () => {
    const wrapper = mount(SButton, {
      propsData: {
        primary: true,
      },
      slots: {
        default: 'Primary button',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders small button', () => {
    const wrapper = mount(SButton, {
      propsData: {
        small: true,
      },
      slots: {
        default: 'Small button',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
