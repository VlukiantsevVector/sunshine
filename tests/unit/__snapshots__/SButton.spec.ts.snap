// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SButton renders normal button 1`] = `<button class="s-button s-button--normal"><span tabindex="-1" class="s-button__content"> <span class="flex-grow">Normal button</span> </span></button>`;

exports[`SButton renders primary button 1`] = `<button class="s-button s-button--primary"><span tabindex="-1" class="s-button__content"> <span class="flex-grow">Primary button</span> </span></button>`;

exports[`SButton renders small button 1`] = `<button class="s-button s-button--small s-button--normal"><span tabindex="-1" class="s-button__content"> <span class="flex-grow">Small button</span> </span></button>`;
